address/address.go

```diff
@@ -35,6 +38,9 @@ func EIP55Checksum(unchecksummed string) (string, error) {
 
 	result := v
 	for i := 0; i < len(result); i++ {
+		if i/2 >= len(hash) {
+			return "", ErrInvalidInput
+		}
```

Comment:
```suggestion
		if len(result) / 2 >= len(hash) {
			return "", ErrInvalidInput
		}
```
and put it before the loop?